Miscellaneous/docker/full_stack_example/docker-compose-linux.yml

```diff
@@ -0,0 +1,223 @@
+---
+version: "2.1"
+services:
+  # The environment variable "ELASTIC_VERSION" is used throughout this file to
+  # specify the version of the images to run. The default is set in the
+  # '.env' file in this folder. It can be overridden with any normal
+  # technique for setting environment variables, for example:
+  #
+  #   ELASTIC_VERSION=5.5.1 docker-compose up
+  #
+  # Additionally, the user can control:
+  #   * the total memory assigned to the ES container through the variable ES_MEM_LIMIT e.g. ES_MEM_LIMIT=2g
+  #   * the memory assigned to the ES JVM through the variable ES_JVM_HEAP e.g. ES_JVM_SIZE=1024m
+  #   * the password used for the elastic, logstash_system and kibana accounts through the variable ES_PASSWORD
+  #   * the mysql root password through the var MYSQL_ROOT_PASSWORD
+  #   * the default index pattern used in kibana via the var DEFAULT_INDEX_PATTERN
+  #   * the ES heap size through tt
+  # REF: https://docs.docker.com/compose/compose-file/#variable-substitution
+  #
+
+  #Elasticsearch container
+  elasticsearch:
+    container_name: elasticsearch
+    hostname: elasticsearch
+    image: "docker.elastic.co/elasticsearch/elasticsearch:${ELASTIC_VERSION}"
+    environment:
+      - http.host=0.0.0.0
+      - transport.host=127.0.0.1
+      - bootstrap.memory_lock=true
+      - "ES_JAVA_OPTS=-Xms${ES_JVM_HEAP} -Xmx${ES_JVM_HEAP}"
+    mem_limit: ${ES_MEM_LIMIT}
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
+    volumes:
+      - ./config/elasticsearch/elasticsearch.yml:/usr/share/elasticsearch/elasticsearch.yml
+      - esdata:/usr/share/elasticsearch/data
+    #Port 9200 is available on the host
+    ports: ['127.0.0.1:9200:9200']
+    #Healthcheck to confirm availability of ES. Other containers wait on this.
+    healthcheck:
+      test: ["CMD", "curl","-s" ,"-f", "-u", "elastic:${ES_PASSWORD}", "http://localhost:9200/_cat/health"]
```

Comment:
I don't like this as a "healthcheck" for the container. 

The health of the container is different than the health of the Elastic Cluster IMO. 

I think the container should be considered healthy if you curl `/` you get back the default json object with `"tag": "You know for Search"` or what ever it is.